src/main/java/duke/dukehelper/Routine.java

```diff
@@ -0,0 +1,125 @@
+    public static boolean echoCommand = false;
+    public static boolean autoSave = true;
+    public static Scanner in = new Scanner(System.in);
+    public static String input;
+
+    private static Packet helperPacket;
+
+
+    /**
+     * Subroutine to handle nested command handling.
+     * Useful if another command needs to be called from the current command.
+     * @param command nested command in handleCommand()
+     * @param isReply
+     * @return
+     */
+    public static void replyRoutine(Constants.Command command, boolean isReply) {
+        Packet packet = null;
+        commandToReply = command;
+        while (commandToReply != null) {
+            CommandHandler.handleCommand(Constants.Command.PROMPT_INPUT);
+            input = in.nextLine();
+            packet = Parser.parseInput(input);
+            if (echoCommand) {
+                CommandHandler.handleCommand(Constants.Command.ECHO, packet);
+            }
+            commandToReply = CommandHandler.handleCommand(commandToReply, packet, true, isReply);
+            isReply = true;
+        }
+    }
+
+    /**
+     * Main subroutine to receive inputs from users.
+     * @param args passed from main()
+     */
+    public static void queryRoutine(String[] args) {
+        // Passed from main.
+        // If true, command inputs will be echoed to console.
+        // Mainly cosmetic in purpose for printing output from script.
+        echoCommand = (args.length > 0) && Boolean.parseBoolean(args[0]);
+
+        System.out.println(Constants.DEFAULT_SAVE_PATH);
+        Packet packet = null;
+        boolean continueQuery = true;
+
+        UiManager.drawPartition();
+        CommandHandler.handleCommand(Constants.Command.HELLO);
+        if (autoSave) {
+            helperPacket = Parser.parseInput("load /name lastSave");
+            CommandHandler.handleCommand(Constants.Command.LOAD_FILE, helperPacket);
+        }
+        while (continueQuery){
+            if (commandToReply != null) {
+                replyRoutine(commandToReply, true);
+            } else {
+                CommandHandler.handleCommand(Constants.Command.PROMPT_INPUT);
+                input = in.nextLine();
+                packet = Parser.parseInput(input);
+
+                if (echoCommand) {
+                    CommandHandler.handleCommand(Constants.Command.ECHO, packet);
+                }
+
+                switch (packet.getPacketType()) {
+                case "bye":
+                    if (autoSave) {
+                        CommandHandler.handleCommand(Constants.Command.REFRESH_FILE);
+                        helperPacket = Parser.parseInput("save /name lastSave");
+                        CommandHandler.handleCommand(Constants.Command.SAVE_FILE, helperPacket);
+                    }
+                    commandToReply = CommandHandler.handleCommand(Constants.Command.BYE);
+                    continueQuery = false;
+                    break;
+                case "list":
```

Comment:
This switch-case statement is exemplary - each case doesn't have too many lines, making it very readable. Perhaps this standard can be applied to CommandHandler.java?

Reply:
Noted! I will see what I can do